crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -390,32 +390,42 @@ impl GeneratedAcir {
             "range check with bit size of the prime field is not implemented yet"
         );
 
-        let mut aof = a + offset;
+        let mut lhs_offset = lhs + offset;
 
-        if b.is_const() && b.q_c.fits_in_u128() {
-            let f = if *offset == Expression::one() {
-                aof = a.clone();
-                assert!(b.q_c.to_u128() >= 1);
-                b.q_c.to_u128() - 1
+        // Optimisation when rhs is const
+        if rhs.is_const() && rhs.q_c.fits_in_u128() {
+            // We try to move the offset to rhs
+            let rhs_offset = if *offset == Expression::one() {
+                lhs_offset = lhs.clone();
+                assert!(rhs.q_c.to_u128() >= 1);
+                rhs.q_c.to_u128() - 1
             } else {
-                b.q_c.to_u128()
+                rhs.q_c.to_u128()
             };
+            // we now have lhs+offset <= rhs <=> lhs_offset <= rhs_offset
 
-            if f < 3 {
```

Comment:
Could we get rid of this optimization and go with shorter more readable code?

Reply:
Done